### PR TITLE
breaking: Remove deprecated nodeVersion configuration option

### LINKED ISSUE #27016: Cypress uses the system Node.js when `nodeVersion` is set to `bundled`
### Current behavior

When the `nodeVersion` configuration option is set to `bundled`, Cypress uses the system Node.js to execute `setupNodeEvents` and tasks

### Desired behavior

I see two ways of fixing this bug:
1. Fix the `nodeVersion` configuration option so that `bundled` works as it used to, and Cypress uses the bundled Node.js to execute `setupNodeEvents` and tasks
2. Remove the `nodeVersion` configuration option altogether, as has been the stated intention since Cypress 9.0.0

My team and I would prefer solution 1, as stated and explained in #18882 . However, that issue didn't gain a lot of traction, so if you choose solution 2 I can understand that

### Test code to reproduce

cypress.config.js:
```js
module.exports = {
  e2e: {
    nodeVersion: 'bundled',
    setupNodeEvents(on, config) {
      console.log(process.version)
      on('task', {
        logNodeVersion() {
          console.log(process.version)
          return true
        }
      })
    },
  },
}
```

spec.cy.js:
```js
it('', () => {
  cy.task('logNodeVersion')
})
```

Make sure you have a system Node.js with a different version than the bundled Node.js. Note the version logged in the terminal when you select E2E testing and when you run spec.cy.js

### Cypress Version

12.14.0

### Node version

v18.16.0

### Operating System

Windows 10.0.19044.2965

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by mjhenkes**
- Closes #27016

### Additional details

Removes the 'nodeVersion' configuration option.

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ _Released 08/1/2023 (PENDING)_
 
 - The [`cy.readFile()`](/api/commands/readfile) command is now retry-able as a [query command](https://on.cypress.io/retry-ability). This should not affect any tests using it; the functionality is unchanged. However, it can no longer be overwritten using [`Cypress.Commands.overwrite()`](/api/cypress-api/custom-commands#Overwrite-Existing-Commands). Addressed in [#25595](https://github.com/cypress-io/cypress/pull/25595).
 
+- The deprecated configuration option, `nodeVersion` has bee removed. Addresses [#27016](https://github.com/cypress-io/cypress/issues/27016).
```
Comment (MikeMcC399) [inline cli/CHANGELOG.md]:
```suggestion
- The deprecated configuration option, `nodeVersion` has been removed. Addresses [#27016](https://github.com/cypress-io/cypress/issues/27016).
```

File: npm/webpack-dev-server/src/helpers/nextHandler.ts [hunk 2]
```diff
@@ -239,6 +239,7 @@ function checkSWC (
 
   // "resolvedNodePath" is only set when using the user's Node.js, which is required to compile Next.js with SWC optimizations
   // If it is not set, they have either explicitly set "nodeVersion" to "bundled" or are are using Cypress < 9.0.0 where it was set to "bundled" by default
+  // @ts-expect-error nodeVersion has been removed as of 13.0.0 however this plugin can be used with many version of cypress
```
Comment (MikeMcC399) [inline npm/webpack-dev-server/src/helpers/nextHandler.ts]:
```suggestion
  // @ts-expect-error nodeVersion has been removed as of 13.0.0 however this plugin can be used with many versions of cypress
```

File: cli/CHANGELOG.md (hunk 1 above)
Comment (mjhenkes) [reply cli/CHANGELOG.md]: 🐝

File: packages/config/src/project/utils.ts [hunk 3]
```diff
@@ -250,7 +250,7 @@ const convertRelativeToAbsolutePaths = (projectRoot: string, obj: Config) => {
 // instead of the built-in Node process, specify a path to 3rd party Node
 export const setNodeBinary = (obj: Config, userNodePath?: string, userNodeVersion?: string) => {
   // if execPath isn't found we weren't executed from the CLI and should used the bundled node version.
-  if (userNodePath && userNodeVersion && obj.nodeVersion !== 'bundled') {
+  if (userNodePath && userNodeVersion) {
```
Comment (mjhenkes) [inline packages/config/src/project/utils.ts:253]: We could still end up using 'bundled' if user node isn't found.

File: packages/errors/src/errors.ts [hunk 4]
```diff
@@ -1022,14 +1022,6 @@ export const AllCypressErrors = {
 
         Please verify that this is the path to a valid, unpacked WebExtension.`
   },
-  COULD_NOT_FIND_SYSTEM_NODE: (nodeVersion: string) => {
```
Comment (mjhenkes) [inline packages/errors/src/errors.ts:1025]: This error was dead code

**Review by chrisbreiding — APPROVED** [review]

**Review by AtofStryker — APPROVED** [review]

**Pull request merged**